### PR TITLE
added runge-kutta-gills method to python/maths

### PR DESCRIPTION
**Pull request by iamrknain**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: maths/runge_kutta_gills.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+"""
+Use the Runge-Kutta-Gill's method of order 4 to solve Ordinary Differential Equations.
+Author : Ravi Kumar
```
Comment (cclauss) [inline maths/runge_kutta_gills.py:3]:
```suggestion
https://en.wikipedia.org/wiki/Runge%E2%80%93Kutta_methods
Author : Ravi Kumar
```

**Comment by cclauss** [comment]
I wonder:
1. Should these go into the arithmatic_analysis directory instead of the generic `math` directory?
2. Should the `arithmatic_analysis` directory be renamed to `numerical_analysis`?
    *https://en.wikipedia.org/wiki/Numerical_analysis
3 Should we combine this and two existing Runge-Kutta algorithms so visitors can easily study them together?

File: maths/runge_kutta_gills.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+"""
+
+from collections.abc import Callable
+
+import numpy as np
+
+from math import sqrt
```
Comment (cclauss) [inline maths/runge_kutta_gills.py:10]:
```suggestion

from collections.abc import Callable
from math import sqrt

import numpy as np
```

File: maths/runge_kutta_gills.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+Author : Ravi Kumar
+"""
+from collections.abc import Callable
+from math import sqrt
+
+import numpy as np
+
+
+def runge_kutta_gills(
+    func: Callable,
```
Comment (tianyizheng02) [inline maths/runge_kutta_gills.py:14]:
```suggestion
    func: Callable[[float, float], float],
```
Please annotate the parameter and output types

File: maths/runge_kutta_gills.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            "The final value of x must be greater than initial value of x."
+        )
+
+    if step_size <= 0:
+        raise ValueError("Step size must be positive.")
+
+    n = int((x_final - x_initial) / step_size)
+    y = np.zeros(
+        (n + 1),
+    )
```
Comment (tianyizheng02) [inline maths/runge_kutta_gills.py:72]: Why not this?
```suggestion
    y = np.zeros(n + 1)
```

Also, is there a reason why you're using a numpy array as opposed to a built-in Python list?

File: maths/runge_kutta_gills.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    x = np.zeros(n + 1)
+    y[0] = y_initial
+    x[0] = x_initial
+    for i in range(n):
```
Comment (tianyizheng02) [inline maths/runge_kutta_gills.py:76]: Could you have this for-loop iterate directly over x values instead?
```suggestion
    for x in range(x_initial, x_final, step_size):
```
Or something along those lines. That way you don't need to manually increment x, nor do you need store the x values in a list.

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**